deployer/hub.py

```diff
@@ -426,14 +427,30 @@ def deploy(self, auth_provider, secret_key, skip_hub_health_test=False):
         if self.spec["template"] == "daskhub":
             subprocess.check_call(["helm", "dep", "up", "daskhub"])
         os.chdir("..")
+        
+        # Check if this cluster has any secret config. If yes, read it in.
+        secret_config_path = Path(os.getcwd()) / "secrets/config/hubs" / f'{self.cluster.spec["name"]}.cluster.yaml'
+
+        if os.path.exists(secret_config_path):
+            with decrypt_file(secret_config_path) as decrypted_file_path:
+                with open(decrypted_file_path) as f:
+                    secret_config = yaml.load(f)
+
+            hubs = secret_config["hubs"]
+            secret_hub_config = next((hub for i, hub in enumerate(hubs) if hubs[i]["name"] == self.spec["name"]), None)
+            secret_hub_config = secret_hub_config["config"]
```

Comment:
If `secret_hub_config` is `None` (ie. putting the wrong name?), 441 will trigger an exception, right?

Reply:
Good question, I don't know! I copied this line from here:

https://github.com/2i2c-org/pilot-hubs/blob/e96e7bcded187870dc2e07d6626de8a12586ed32/deployer/__main__.py#L89-L96

And since that part handles sorting the hub names and parsing them to `deploy`, I don't think the `None` would ever actually be triggered here.

However, syntactically, I now think `None` should be something like `{"config": {}}` in order to match what is being output by the `else` clause.

Reply:
> However, syntactically, I now think None should be something like {"config": {}}

Yep, I was expecting something like that.

Reply:
Done in [`c937694`](https://github.com/2i2c-org/pilot-hubs/pull/706/commits/c93769442483127f4fd7a5cbe5ba98860ae6217d)